File: minyoung/BOJ_10773.java
import java.util.StringTokenizer;
import java.util.Stack;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.BufferedWriter;

public class Main
{
	public static int K;
	public static void main (String[] args) throws java.lang.Exception
	{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		Stack <Integer> stack = new Stack<>();
		
		
		K = Integer.parseInt(br.readLine());
		
		for(int i=0;i<K;i++){

			String str = br.readLine();
			
			if(str.equals("0")){
				stack.pop();
			}
			else{
				stack.push(Integer.parseInt(str));
			}
			
		}
	
		int sum=0;
		while(!stack.empty()){
			sum=sum+stack.pop();
		}
		
		bw.write(String.valueOf(sum));
		bw.flush();
	}
}

File: hyerang0125/0x05/boj_10773.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int k, num, sum;
    stack<int> s;

    cin >> k;
    for(int i=0; i<k; i++){
        cin >> num;
        if(num == 0){
            if(s.empty()) continue;
            else s.pop();
        }
        else{
            s.push(num);
        }
    }

    sum = 0;
    while(!s.empty()){
        sum += s.top();
        s.pop();
    }

    cout << sum << endl;

    return 0;
}